build/tools/avalanchejs/create-an-asset-on-the-x-chain.md

```diff
@@ -1,112 +1,151 @@
+
+Now we need to create an empty array for the `SECPMintOutput` which we're going to create. We also need a `threshold` and `locktime` for the outputs which we're going to create. Each X-Chain transaction can accepts a 256 byte `memo` of arbitrary data.
+
+```ts
+const outputs: SECPMintOutput[] = []
+const threshold: number = 1
+const locktime: BN = new BN(0)
+const memo: Buffer = bintools.stringToBuffer("AVM utility method buildCreateAssetTx to create an ANT")
 ```
 
 ## Describe the new asset
 
 The first step in creating a new asset using AvalancheJS is to determine the qualities of the asset. We will give the asset a name, a ticker symbol, as well as a denomination.
 
-```text
-// Name our new coin and give it a symbol
-let name = "Rickcoin is the most intelligent coin";
-let symbol = "RICK";
-
+```ts
+const name: string = "TestToken"
+const symbol: string = "TEST"
 // Where is the decimal point indicate what 1 asset is and where fractional assets begin
 // Ex: 1 AVAX is denomination 9, so the smallest unit of AVAX is nanoAVAX (nAVAX) at 10^-9 AVAX
-let denomination = 9;
+const denomination: number = 3
 ```
 
-## Creating the initial state
+## Set up async/await
 
-We want to mint an asset with 400 coins to all of our managed keys, 500 to the second address we know of, and 600 to the second and third address. This sets up the state that will result from the Create Asset transaction.
+The remaining code will be encapsulated by this `main` function so that we can use the `async` / `await` pattern.
 
-_Note: This example assumes we have the keys already managed in our X-Chain’s Keychain._
+```ts
+const main = async (): Promise<any> => {
+}
+main()
+```
 
-```text
-let addresses = xchain.keyChain().getAddresses();
+## Fetch the UTXO
 
-// Create outputs for the asset's initial state
-let secpOutput1 = new SECPTransferOutput(new BN(400), new BN(400), 1, addresses);
-let secpOutput2 = new SECPTransferOutput(new BN(500), new BN(400), 1, [addresses[1]]);
-let secpOutput3 = new SECPTransferOutput(new BN(600), new BN(400), 1, [addresses[1], addresses[2]]);
+Pass the `xAddressStrings` to `xchain.getUTXOs` to fetch the UTXO.
 
-// Populate the initialStates with the outputs
-let initialState = new InitialStates();
-initialState.addOutput(secpOutput1);
-initialState.addOutput(secpOutput2);
-initialState.addOutput(secpOutput3);
+```ts
+  const avmUTXOResponse: iAVMUTXOResponse = await xchain.getUTXOs(xAddressStrings)
+  const utxoSet: UTXOSet = avmUTXOResponse.utxos
 ```
 
-## Creating the signed transaction
-
-Now that we know what we want an asset to look like, we create an output to send to the network. There is an AVM helper function `buildCreateAssetTx()` which does just that.
+## Creating the initial state
 
-```text
-// Fetch the UTXOSet for our addresses
-let utxos = await xchain.getUTXOs(addresses);
+We want to mint an asset with 507 coins to of our managed key. This sets up the state that will result from the Create Asset transaction.
```

Comment:
`We want to mint an asset with 507 coins to of our managed key` --> `We want to mint an asset with 507 units of the asset held by the managed key.`